### PR TITLE
Ywallet app submission

### PR DESCRIPTION
**Pull request by hhanh00**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: me.hanh.zwallet.Ywallet.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        sha256: 4b6cbf8056cc2dbbdebd70d5730fb864b12074a9f6d184adf7d2b1939327a9a0
+        strip-components: 0
+    build-commands:
+      - mkdir /app/bin
+      - cp -r * /app/bin
+  - name: warp_api_ffi
+    buildsystem: simple
+    sources:
+      - type: file
+        url: https://github.com/hhanh00/zwallet/releases/download/v1.2.15/libwarp_api_ffi.so
```
Comment (hfiguiere) [inline me.hanh.zwallet.Ywallet.yml:28]: Given that both aarch64 and x86_64 flatpak are built I expect this to be a problem.

File: me.hanh.zwallet.Ywallet.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+runtime-version: "21.08"
+sdk: org.freedesktop.Sdk
+command: ywallet
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --filesystem=xdg-data/databases:create
```
Comment (hfiguiere) [inline me.hanh.zwallet.Ywallet.yml:12]: why do you need this?

File: me.hanh.zwallet.Ywallet.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+id: me.hanh.zwallet.Ywallet
+runtime: org.freedesktop.Platform
+runtime-version: "21.08"
```
Comment (hfiguiere) [inline me.hanh.zwallet.Ywallet.yml:3]:
```suggestion
runtime-version: "22.08"
```

File: me.hanh.zwallet.Ywallet.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
+id: me.hanh.zwallet.Ywallet
```
Comment (hfiguiere) [inline me.hanh.zwallet.Ywallet.yml:1]: `hanh.me` doesn't resolve. This doesn't meet the app-id requirements.

File: me.hanh.zwallet.Ywallet.yml (hunk 1 above)
Comment (hhanh00) [reply me.hanh.zwallet.Ywallet.yml:31]: Added only_arches

File: me.hanh.zwallet.Ywallet.yml (hunk 2 above)
Comment (hhanh00) [reply me.hanh.zwallet.Ywallet.yml:13]: Added comment

File: me.hanh.zwallet.Ywallet.yml (hunk 3 above)
Comment (hhanh00) [reply me.hanh.zwallet.Ywallet.yml]: Changed as suggested

File: me.hanh.zwallet.Ywallet.yml (hunk 4 above)
Comment (hhanh00) [reply me.hanh.zwallet.Ywallet.yml]: Changed to `ywallet.app`

File: me.hanh.zwallet.Ywallet.yml (hunk 2 above)
Comment (hfiguiere) [reply me.hanh.zwallet.Ywallet.yml:13]: the comment doesn't answer the question

File: me.hanh.zwallet.Ywallet.yml (hunk 1 above)
Comment (hfiguiere) [reply me.hanh.zwallet.Ywallet.yml:31]: it doesn't do what you think it does. it just skips.

File: me.hanh.zwallet.Ywallet.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
+id: app.ywallet.Ywallet
```
Comment (hfiguiere) [inline me.hanh.zwallet.Ywallet.yml:1]: the manifest file name doesn't match.

File: me.hanh.zwallet.Ywallet.yml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      - type: file
+        path: ywallet.desktop
+      - type: file
+        path: ywallet.metainfo.xml
+      - type: file
+        path: icon.png
+    build-commands:
+      - install -Dp -m 644 icon.png /app/share/icons/hicolor/256x256/apps/me.hanh.zwallet.Ywallet.png
+      - install -Dp -m 644 ywallet.desktop /app/share/applications/me.hanh.zwallet.Ywallet.desktop
+      - install -Dp -m 644 ywallet.metainfo.xml /app/share/metainfo/me.hanh.zwallet.Ywallet.metainfo.xml
```
Comment (hfiguiere) [inline me.hanh.zwallet.Ywallet.yml:47]: and these are installed with the wrong name.

File: ywallet.desktop [hunk 7]
```diff
@@ -0,0 +1,7 @@
+[Desktop Entry]
+Name=YWallet
+Comment=Light Wallet for Ycash and Zcash
+Exec=ywallet
+Type=Application
+Icon=me.hanh.zwallet.Ywallet
```
Comment (hfiguiere) [inline ywallet.desktop:6]: wrong icon name

File: ywallet.metainfo.xml [hunk 8]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+    <id>me.hanh.zwallet.Ywallet</id>
```
Comment (hfiguiere) [inline ywallet.metainfo.xml:3]: wrong app-id

**Comment by hfiguiere** [comment]
have the desktop, icon and appdata file been submitted upstream? a Linux app without a .desktop and icon is a bug.

**Comment by hfiguiere** [comment]
bot, build app.ywallet.Ywallet

File: app.ywallet.Ywallet.yml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+sdk: org.freedesktop.Sdk
+command: ywallet
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  # Location of SQLite database file for user account data & blockchain state
+  - --filesystem=xdg-data/databases:create
```
Comment (hfiguiere) [inline app.ywallet.Ywallet.yml:13]: the build fail because of this.

**Comment by hhanh00** [comment]
I don't understand the error: finish-args-unnecessary-xdg-data-access
Is there any documentation on it?

**Comment by hhanh00** [comment]
App works fine without it. So I removed it.

**Comment by hfiguiere** [comment]
bot, build app.ywallet.Ywallet

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/app.ywallet.Ywallet.appdata.xml: FAILED:
• style-invalid         : Not enough <screenshot> tags, minimum is 1
Validation of files failed
```

**Comment by hhanh00** [comment]
bot, build app.ywallet.Ywallet

**Pull request closed without merging**